### ISSUE TITLE
`itkDebugMacro` errors in `itk::GaussianOperator`

### ISSUE THREAD

**Issue by tbirdso** [InsightSoftwareConsortium/ITK#3430]
### Description

`this->GetDebug()` is invalid when used in `itkDebugMacro` in `itk::GaussianOperator`. The debug statement was reduced from a warning statement in [https://github.com/InsightSoftwareConsortium/ITK/pull/3423](https://github.com/InsightSoftwareConsortium/ITK/pull/3423).

### Steps to Reproduce

[CDash Expected Nightly Builds](https://open.cdash.org/viewBuildError.php?buildid=7917060)

### Expected behavior

No errors

### Actual behavior

```none
In file included from /Users/builder/external[Modules/Core/ImageFunction/test/itkNeighborhoodOperatorImageFunctionTest.cxx:22](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/ImageFunction/test/itkNeighborhoodOperatorImageFunctionTest.cxx#L22):
In file included from /Users/builder/external[Modules/Core/Common/include/itkGaussianOperator.h:192](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkGaussianOperator.h#L192):
/Users/builder/external[Modules/Core/Common/include/itkGaussianOperator.hxx:55](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkGaussianOperator.hxx#L55):7: error: no member named 'GetDebug' in 'itk::GaussianOperator<float, 3, itk::NeighborhoodAllocator<float> >'
      itkDebugMacro(<< "Kernel size has exceeded the specified maximum width of " << m_MaximumKernelWidth
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[CTest: warning matched] /Users/builder/external[Modules/Core/Common/include/itkMacro.h:489](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkMacro.h#L489):17: note: expanded from macro 'itkDebugMacro'
      if (this->GetDebug() && itk::Object::GetGlobalWarningDisplay())          \
          ~~~~  ^
[CTest: warning matched] /Users/builder/external[Modules/Core/ImageFunction/test/itkNeighborhoodOperatorImageFunctionTest.cxx:59](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/ImageFunction/test/itkNeighborhoodOperatorImageFunctionTest.cxx#L59):21: note: in instantiation of member function 'itk::GaussianOperator<float, 3, itk::NeighborhoodAllocator<float> >::GenerateCoefficients' requested here
  auto * oper = new NeighborhoodOperatorType;
                    ^
1 error generated.
```

### Reproducibility

Linux CI passed but expected nightly builds fail

### Versions

`master`

### Environment

Linux, macOS

### Additional Information

Need to investigate where `this->GetDebug` is commonly defined, likely outside of `itk::GaussianOperator` inheritance hierarchy. Mostly likely resolution is to 1) remove printout entirely from `itk::GaussianOperator` and 2) document in `itkDebugMacro` that the macro is only valid to be used in classes inheriting from a certain base.

**Comment by tbirdso** [comment]
Investigation notes so far:
- `m_Debug` is defined in `itk::Object`
- `itk::GaussianOperator` inherits from `itk::Neighborhood` at its base and does not share an inheritance hierarchy with `itk::Object` or `itk::LightObject`. I'm not sure that I fully understand the advantage of having it as a separate hierarchy.
- Several resolutions are available here:
  - Have `itk::Neighborhood` inherit from `itk::Object` -> would add support for common mechanisms such as reference counting, debug printouts, and observers, but could have unintended consequences on lightweight neighborhood operations
  - Add `Set/GetDebug` methods to `itk::GaussianOperator` directly -> less scalable but probably the best resolution for this problem, since other neighborhoods do not seem to need to have debug mechanisms

**Comment by dzenanz** [comment]
I think this was closed due to automation misinterpreting "partly addresses".

**Comment by dzenanz** [comment]
I am now caught up with. This was fixed in #3432.
